src/windows/wslaservice/exe/WSLAContainer.cpp

```diff
@@ -324,6 +323,31 @@ std::vector<std::string> WSLAContainer::PrepareNerdctlCreateCommand(const WSLA_C
     args.push_back("create");
     args.push_back("--name");
     args.push_back(options.Name);
+
+    switch (options.ContainerNetwork.ContainerNetworkType)
+    {
+    case WSLA_CONTAINER_NETWORK_HOST:
+        args.push_back("--net=host");
+        break;
+    case WSLA_CONTAINER_NETWORK_NONE:
+        args.push_back("--net=none");
+        break;
+    case WSLA_CONTAINER_NETWORK_BRIDGE:
+        args.push_back("--net=bridge");
+        break;
+    // TODO: uncomment and implement when we have custom networks
+    // case WSLA_CONTAINER_NETWORK_CUSTOM:
+    //     args.push_back(std::format("--net={}", options.ContainerNetwork.ContainerNetworkName));
+    //     break;
+    default:
+        THROW_HR_MSG(
+            HRESULT_FROM_WIN32(E_INVALIDARG),
```

Comment:
E_INVALIDARG is already an HRESULT, so `HRESULT_FROM_WIN32` should be removed here (otherwise we're going to return an invalid error code)